File: web/src/test/java/no/nav/modiapersonoversikt/service/unleash/strategier/IsNotProdStrategyTest.java
package no.nav.modiapersonoversikt.service.unleash.strategier;

import org.junit.jupiter.api.Test;

import java.util.HashMap;

import static no.nav.common.test.SystemProperties.setTemporaryProperty;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

class IsNotProdStrategyTest {

    final IsNotProdStrategy strategy = new IsNotProdStrategy();
    private final String ENVIRONMENT_PROPERTY = "APP_ENVIRONMENT_NAME";

    @Test
    void setIsNotProdStrategyTest(){
        assertIsEnabled("p", false);
        assertIsEnabled("t1", true);
        assertIsEnabled("q0", true);
        assertIsEnabled("u10", true);
        assertIsEnabled("local", true);
        assertIsEnabled("", true);
        assertIsEnabled(null, true);
    }

    private void assertIsEnabled(String actualEnvironment, boolean expectedStatus){
        setTemporaryProperty(ENVIRONMENT_PROPERTY, actualEnvironment, () -> {
            HashMap<String, String > params = new HashMap<>();
            assertThat(strategy.isEnabled(params), is(expectedStatus));
        });
    }

}

File: web/src/test/java/no/nav/modiapersonoversikt/service/unleash/strategier/IsNotProdStrategyTest.kt
package no.nav.modiapersonoversikt.service.unleash.strategier

import no.nav.modiapersonoversikt.service.unleash.strategier.StrategyUtils.ENVIRONMENT_PROPERTY
import no.nav.personoversikt.common.test.testenvironment.TestEnvironment
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

internal class IsNotProdStrategyTest {
    val isNotProdStrategy = IsNotProdStrategy()

    @ParameterizedTest
    @MethodSource("testcases")
    internal fun `should evaluate strategy`(
        simulatedEnv: String?,
        expected: Boolean,
    ) {
        TestEnvironment.withEnvironment(mapOf(ENVIRONMENT_PROPERTY to simulatedEnv)) {
            assertThat(isNotProdStrategy.isEnabled(null)).isEqualTo(expected)
        }
    }

    companion object {
        @JvmStatic
        fun testcases() =
            listOf(
                strategyEnabled(simulatedEnv = "t1"),
                strategyEnabled(simulatedEnv = "q0"),
                strategyEnabled(simulatedEnv = "u10"),
                strategyEnabled(simulatedEnv = "local"),
                strategyEnabled(simulatedEnv = ""),
                strategyEnabled(simulatedEnv = null),
                strategyDisabled(simulatedEnv = "p"),
            )

        private fun strategyEnabled(simulatedEnv: String?): Arguments = Arguments.of(simulatedEnv, true)

        private fun strategyDisabled(simulatedEnv: String?): Arguments = Arguments.of(simulatedEnv, false)
    }
}
